core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -346,6 +354,7 @@ private[spark] class TaskSchedulerImpl(
     // Build a list of tasks to assign to each worker.
     val tasks = shuffledOffers.map(o => new ArrayBuffer[TaskDescription](o.cores / CPUS_PER_TASK))
     val availableCpus = shuffledOffers.map(o => o.cores).toArray
+    val availableSlots = shuffledOffers.map(o => o.cores / CPUS_PER_TASK).sum
```

Comment:
`availableCpus.sum / CPUS_PER_TAS`?

Reply:
In case `o.cores` not divisible by `CPUS_PER_TASK`, it shall produce larger available slot numbers than reality if we first sum all availableCpus then divide by `CPUS_PER_TASK`.